dstep/translator/MacroDefinition.d

```diff
@@ -1188,7 +1221,417 @@ Expression parseCondExpr(ref TokenRange tokens, bool[string] table)
     return expr;
 }
 
-Expression parseExpr(ref TokenRange tokens, bool[string] table)
+bool parseBasicSpecifier(ref TokenRange tokens, ref string spelling, Cursor[string] table)
+{
+    import std.meta : AliasSeq;
+
+    alias specifiers = AliasSeq!(
+        "void",
+        "char",
+        "short",
+        "int",
+        "long",
+        "float",
+        "double",
+        "signed",
+        "unsigned",
+        // "__complex__", TBD
+        // "_Complex", TBD
+        "bool",
+        "_Bool");
+
+    if (accept!(specifiers)(tokens, spelling))
+    {
+        return true;
+    }
+
+    return false;
+}
+
+bool parseRecordSpecifier(ref TokenRange tokens, ref Type type, Cursor[string] table)
+{
+    auto local = tokens;
+    string spelling;
+
+    if (accept!("struct")(local, TokenKind.keyword) &&
+        acceptIdentifier(local, spelling))
+    {
+        if (auto ptr = ("struct " ~ spelling in table))
+        {
+            type.kind = CXTypeKind.CXType_Record;
+            type.spelling = spelling;
+            tokens = local;
+            return true;
+        }
+    }
+    else if (accept!("union")(local, TokenKind.keyword) &&
+        acceptIdentifier(local, spelling))
+    {
+        if (auto ptr = ("union " ~ spelling in table))
+        {
+            type.kind = CXTypeKind.CXType_Record;
+            type.spelling = spelling;
+            tokens = local;
+            return true;
+        }
+    }
```

Comment:
These two if statements are exactly the same, except for the keyword `struct/union`. If you add an additional variable, `keywordType` or similar and pass that in to the `accept` function and two template arguments `struct` and `union` then `keywordType` will contain which keyword that actually match on return?